jsdom-monkeypatch.js

```diff
@@ -0,0 +1,308 @@
+  // Explicitly whitelist the "autolink" script of the shadow DOM spec which
+  // is needed to initialize respecConfig
+  const respecUrl = 'https://www.w3.org/Tools/respec/respec-w3c-common';
+  function getUrlToFetch() {
+    let referrer = options.referrer;
+    if (!referrer.endsWith('/')) {
+      referrer = referrer.substring(0, referrer.lastIndexOf('/') + 1);
+    }
+    if (/\/respec[\/\-]/i.test(url.path)) {
+      //console.log(`fetch ReSpec (force latest version)`);
+      return respecUrl;
+    }
+    else if (/\.[^\/\.]+$/.test(url.path) &&
+        !url.path.endsWith('.js') &&
+        !url.path.endsWith('.json')) {
+      //console.log(`fetch not needed for ${url.href} (not a JS/JSON file)`);
+      return null;
+    }
+    else if ((url.pathname === '/webcomponents/assets/scripts/autolink.js') ||
+        (url.href.startsWith(referrer) &&
+          !(/annotate_spec/i.test(url.pathname)) &&
+          !(/expanders/i.test(url.pathname)) &&
+          !(/bug-assist/i.test(url.pathname)) &&
+          !(/dfn/i.test(url.pathname)) &&
+          !(/section-links/i.test(url.pathname)) &&
+          !(/^\/webidl\//i.test(url.pathname)))) {
+      //console.log(`fetch useful script at ${url.href}`);
+      return url.href;
+    }
+    //console.log(`fetch not needed for ${url.href}`);
+    return null;
+  }
+
+  let urlToFetch = getUrlToFetch();
+  if (!urlToFetch) {
+    return callback(null, '');
+  }
+  fetch(urlToFetch, options)
+    .then(response => response.text())
+    .then(data => {
+      if (urlToFetch !== respecUrl) {
+          return data;
+      }
+
+      ////////////////////////////////////////////////////////////
+      // REALLY UGLY CODE WARNING
+      //
+      // Tweak Respec built code so that it can run in JSDOM.
+      //
+      // NB: Some of these lines will just break if Respec build
+      // produces slightly different code, e.g. if variables do
+      // not end up with the same name!
+      ////////////////////////////////////////////////////////////
+
+      // Remove core/highlight module because JSDOM does not yet
+      // support URL.createObjectURL
+      // https://github.com/jsdom/jsdom/issues/1721
+      // Remove core/list-sorter module because JSDOM does not yet
+      // support document.createRange
+      // https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/nodes/Document.webidl#L39
+      ["core/highlight", "core/list-sorter"]
+        .forEach(module => data = data.replace(
+          new RegExp('(define\\(\\s*"profile-w3c-common"\\s*,\\s*\\[[^\\]]+),\\s*"' + module + '"'),
+          '$1'));
+
+      // JSDOM's CSS parser does not quite like uncommon "@" rules
+      // so let's pretend they are just @media rules
+      // https://github.com/jsdom/jsdom/issues/2026
+      // (NB: this replacement is just for convenience, to avoid JSDOM reporting
+      // lengthy errors (including a full dump of the CSS) to stderr
```

Comment:
I think in practice we're not interested in CSS parsing - cf https://github.com/jsdom/jsdom/issues/2005